### PR TITLE
Changed the use of pyglet plot module

### LINKED ISSUE #10243: Run the examples during automated testing or at release
Right now `examples/advanced/autowrap_ufuncify` doesn't run with the development version of SymPy because of backwards incompatible API changes, see #10234. We should figure out how to run these on Travis or simply before each release.

### PR DESCRIPTION
**Pull request by tokencolour**
This example uses, as of now, the old pyglet plotting module
which is not working right now.
Changed so that it uses the new plot API.
Trying to fix #10243

### CODE REVIEW

File: examples/advanced/autowrap_ufuncify.py [hunk 1]
```diff
@@ -46,7 +47,7 @@ def main():
     print("Compiling legendre ufuncs and checking results:")
 
     # Let's also plot the ufunc's we generate
-    plot1 = Plot(visible=False)
+    #plot1 = Plot(visible=False)
```
Comment (moorepants) [inline examples/advanced/autowrap_ufuncify.py]: Don't leave in old commented code.

Comment (tokencolour) [inline examples/advanced/autowrap_ufuncify.py]: I'll change that.

**Comment by moorepants** [comment]
It seems to me you completely removed the purpose of the example. I think the original author was trying to show how to make use of a binary_ufunc in plotting to speed things up. The new plot command is simply plotting the sympy expression. If the new plot command is using a fast numerical evaluation in the background, then maybe this example isn't useful anymore. But my guess is that the plot command uses evalf, so passing in the generated function would make the plotting faster.

**Comment by tokencolour** [comment]
The problem is that I am not able to pass a sympified expression to `plot` method once it is ufuncified. Passing `expr` directly defeats the purpose, but I had no choice. Could you please suggest something?

**Comment by moorepants** [comment]
That sounds like `plot()` may need more than the `evalf` method then. Theorectically, this should work:

```python
expr = x**2
fast_expr = binary_ufunc(expr, x)
plot(fast_expr)
```

**Comment by tokencolour** [comment]
By `binary_ufunc` do you mean the compiled  unevaluated ufunc created by ufuncifying the expr?:
as in 
```
>>> binary_ufunc=ufuncify(x, expr)
>>> fast_expr= binary_ufunc(expr, x)
```

**Comment by tokencolour** [comment]
If that is true, that is where my problem is, binary_ufunc expects arrays. I wish there was a way we could get back to the expression once we ufuncified it.

**Comment by moorepants** [comment]
I don't think you want to get the expression back. You need to have a look at how the `plot` function works and figure out why the generated function doesn't work.

**Comment by moorepants** [comment]
Actually, I think you should just use matplotlib's plot function with the ufuncified expression.

The `binary_function()` creates a SymPy function, not a SymPy expression.

**Comment by tokencolour** [comment]
I'll have to see if matplotlib's plot function will be able to handle a Sympy expression. Plotting a ufuncified function is no help with Sympy's plot.  I had tried this:
```
 g=implemented_function('g', binary_poly)
 plot(g(x))
```
I get a TypeError. Kindly look at https://github.com/sympy/sympy/issues/10243#issuecomment-166097361

**Comment by moorepants** [comment]
You don't want to use a sympy expression, just do:

```python
expr = x**2
fast_expr = ufuncify(x, expr)
import matplotlib.pyplot as plt
plt.plot(range(10), fast_expr(range(10)))
plt.show()
```

**Comment by tokencolour** [comment]
This is working fine! I'll add this. Thank you!

**Comment by tokencolour** [comment]
@moorepants Kindly take a look at this.

File: examples/advanced/autowrap_ufuncify.py [hunk 2]
```diff
@@ -72,11 +72,11 @@ def main():
...
+        plot1 = plt.plot(grid, polyvector, hold=True)
+
 
     print("Here's a plot with values calculated by the wrapped binary functions")
-    plot1.show()
+    plt.show()
 
 if __name__ == '__main__':
-    main()
+    main()
```
Comment (moorepants) [inline examples/advanced/autowrap_ufuncify.py]: Missing end of line.

File: examples/advanced/autowrap_ufuncify.py [hunk 3]
```diff
@@ -26,9 +26,10 @@
     sys.exit("Cannot import numpy. Exiting.")
 
 import mpmath
+import matplotlib.pyplot as plt
```
Comment (moorepants) [inline examples/advanced/autowrap_ufuncify.py]: Since matplotlib is not a sympy dependency, you should do the same thing as numpy above.

**Comment by tokencolour** [comment]
Kindly take a look at this.

**Comment by moorepants** [comment]
Looks good!

**Comment by tokencolour** [comment]
Thank you!

**Pull request merged**